### PR TITLE
1365 daten zum live view count

### LINKED ISSUE #1365: Daten zum Live View Count
(This was mistakenly posted to the TCAs support chanel, not my issue)

Sehr geehrtes TUM Developer Team,

wir haben uns gewundert, wie sich der Live View Count einer gestreamten Vorlesung im Verlauf der Zeit ändert. Die aufwändige Methode, das zu ermitteln, wäre diesen Wert lokal aufzuzeichnen. Andererseits wird die Ermittlung dieses Wertes als serverseitiges Feature genannt. Und da diese Zahl auch jedem angezeigt wird / für jeden sichtbar ist, wäre die Frage, ob Sie diese Daten nachträglich zur Analyse zur Verfügung stellen.

Wenn ja würden wir uns freuen, wenn wir diese Daten genauer anschauen könnten.
Wir wünschen Ihnen noch einen schönen Freitag Nachmittag.

Mit freundlichen Grüßen,
[REDACTED]

### PR DESCRIPTION
**Pull request by SebiWrn**
### Motivation and Context
Fixes #1365 

### Description
Now there's a button on the watch page to view the live stats as well as some other stats about a single lecture but not a whole course

### Steps for Testing
Prerequisites:
- 1 Account (Course Admin)
- 1 Livestream (finished)

1. Log in
2. Navigate to a Livestream
3. Click on the stats icon at the botton

### Screenshots
image

### CODE REVIEW

File: dao/statistics.go [hunk 1]
```diff
@@ -102,6 +134,45 @@ func (d statisticsDao) GetCourseStatsHourly(courseID uint) ([]Stat, error) {
...
+	err := DB.Raw(`SELECT Date_FORMAT(stats.time, "%H:%i") AS x, stats.viewers AS y
+		FROM stats
+			JOIN streams s ON s.id = stats.stream_id
+		WHERE s.course_id = ? AND s.id = ? AND stats.live = 1
+		ORDER BY x;`, courseID, streamID).Scan(&res).Error
+	// err := DB.Raw(`SELECT TIMESTAMPDIFF(MINUTE, s.start, stats.time) AS x, stats.viewers AS y
+	//	FROM stats
+	//		JOIN streams s ON s.id = stats.stream_id
+	//	WHERE s.course_id = ? AND stats.live = 1
+	//	ORDER BY x;`, courseID).Scan(&res).Error
```
Comment (karjo24) [inline dao/statistics.go]: Delete if not used?

File: dao/statistics.go [hunk 2]
```diff
@@ -102,6 +134,45 @@ func (d statisticsDao) GetCourseStatsHourly(courseID uint) ([]Stat, error) {
 	return res, err
 }
 
+// GetCourseStatsWeekdays returns the days and their sum of vod views of a course
```
Comment (karjo24) [inline dao/statistics.go]: Docstring doesn't match function name

File: dao/statistics.go [hunk 3]
```diff
@@ -102,6 +134,45 @@ func (d statisticsDao) GetCourseStatsHourly(courseID uint) ([]Stat, error) {
...
+	err := DB.Raw(`SELECT DAYNAME(stats.time) AS x, SUM(stats.viewers) as y
+		FROM stats
+			JOIN streams s ON s.id = stats.stream_id
+		WHERE (s.course_id = ? OR ? = 0) AND stats.live = 0 AND stats.stream_id = ?
+		GROUP BY DAYOFWEEK(stats.time);`,
+		courseID, courseID, streamID).Scan(&res).Error
+	return res, err
+}
+
+// GetCourseStatsHourly returns the hours with most vod viewing activity of a course
```
Comment (karjo24) [inline dao/statistics.go]: Docstring doesn't match function name.

File: dao/statistics.go [hunk 4]
```diff
@@ -102,6 +134,45 @@ func (d statisticsDao) GetCourseStatsHourly(courseID uint) ([]Stat, error) {
...
 
+// GetCourseStatsWeekdays returns the days and their sum of vod views of a course
+func (d statisticsDao) GetLectureStatsWeekdays(courseID uint, streamID uint) ([]Stat, error) {
+	var res []Stat
+	err := DB.Raw(`SELECT DAYNAME(stats.time) AS x, SUM(stats.viewers) as y
+		FROM stats
+			JOIN streams s ON s.id = stats.stream_id
+		WHERE (s.course_id = ? OR ? = 0) AND stats.live = 0 AND stats.stream_id = ?
+		GROUP BY DAYOFWEEK(stats.time);`,
+		courseID, courseID, streamID).Scan(&res).Error
```
Comment (karjo24) [inline dao/statistics.go:145]: `(s.course_id = ? OR ? = 0)` in `WHERE`-clause unnecessary as `stats.stream_id = ?` implicitly checks for the right course anyways?

**Review by karjo24 — CHANGES_REQUESTED** [review]
Looks good

File: dao/statistics.go (hunk 4 above)
Comment (SebiWrn) [reply dao/statistics.go:145]: I ran into some errors when trying it without this check and it's also good to check if the client knows both the courseID and the streamID

**Review by karjo24 — APPROVED** [review]

**Pull request merged**